lightning/src/routing/router.rs

```diff
@@ -129,6 +130,48 @@ pub trait Router: MessageRouter {
 	) -> Result<Route, LightningError> {
 		self.find_route(payer, route_params, first_hops, inflight_htlcs)
 	}
+
+	/// Creates [`BlindedPath`]s for payment to the `recipient` node. The channels in `first_hops`
+	/// are assumed to be with the `recipient`'s peers. The payment secret and any constraints are
+	/// given in `tlvs`.
+	///
+	/// The default implementation returns two-hop payment paths for each channel in `first_hops`
+	/// with the `recipient` node (i.e., the channel counterparty is the introduction point).
+	fn create_blinded_payment_paths<
+		ES: EntropySource + ?Sized, T: secp256k1::Signing + secp256k1::Verification
+	>(
+		&self, recipient: PublicKey, first_hops: Vec<ChannelDetails>, tlvs: ReceiveTlvs,
+		amount_msats: u64, count: usize, entropy_source: &ES, secp_ctx: &Secp256k1<T>
+	) -> Result<Vec<(BlindedPayInfo, BlindedPath)>, ()> {
+		first_hops.into_iter()
+			.filter(|details| details.counterparty.features.supports_route_blinding())
+			.filter(|details| amount_msats >= details.inbound_htlc_minimum_msat.unwrap_or(0))
+			.filter(|details| amount_msats <= details.inbound_htlc_maximum_msat.unwrap_or(0))
+			.map(|details| {
+				let short_channel_id = details.get_inbound_payment_scid().unwrap();
```

Comment:
I don't think we should unwrap here and below.

Reply:
IIUC, a usable channel will always have these set. I could add debug assertions or additional filters to be safe, if you prefer.

Reply:
Yeah, while unlikely this panic is reachable from the public interface so I think we should filter.